install_opi5.sh

```diff
@@ -4,55 +4,89 @@ if id "$1" >/dev/null 2>&1; then
     echo 'user found'
 else
     echo "creating pi user"
-    useradd pi -b /home
+    useradd pi -m -b /home -s /bin/bash
     usermod -a -G sudo pi
-    mkdir /home/pi
-    chown -R pi /home/pi
 fi
 echo "pi:raspberry" | chpasswd
 
-apt-get update
+apt-get update --quiet
+
+before=$(df --output=used / | tail -n1)
+# clean up stuff
+echo 'Purging snaps'
+# get rid of snaps
+rm -rf /var/lib/snapd/seed/snaps/*
+rm -f /var/lib/snapd/seed/seed.yaml
+apt-get purge --yes --quiet lxd-installer lxd-agent-loader
+apt-get purge --yes --quiet snapd
+
+# remove bluetooth daemon
+apt-get purge --yes --quiet bluez
+
+apt-get --yes --quiet autoremove
+
+after=$(df --output=used / | tail -n1)
+freed=$(( before - after ))
+
+echo "Freed up $freed KiB"
+
+# run Photonvision install script
 wget https://git.io/JJrEP -O install.sh
 chmod +x install.sh
 
-sed -i 's/# AllowedCPUs=4-7/AllowedCPUs=4-7/g' install.sh
+sed -i 's/# AllowedCPUs=4-7/AllowedCPUs=0-7/g' install.sh
 
 ./install.sh -n -q
 rm install.sh
 
+echo "Installing additional things"
 
-# Remove extra packages 
-echo "Purging extra things"
-# apt-get remove -y gdb gcc g++ linux-headers* libgcc*-dev
-# apt-get remove -y snapd
-apt-get autoremove -y
+apt-get install --yes --quiet network-manager net-tools libatomic1
 
+# set NetworkManager as the renderer in cloud-init and static ip
+cp -f ./OPi5_CIDATA/network-config /boot/network-config
 
-echo "Installing additional things"
-sudo apt-get update
-apt-get install -y network-manager net-tools libatomic1
+# set the hostname in cloud-init
+sed -i 's/#hostname:.*/hostname: photonvision/' /boot/user-data
+grep 'hostname:' /boot/user-data
+
+# add run command to disable cloud-init after first boot
+sed -i '$a\\nruncmd:\n- [ touch, /etc/cloud/cloud-init.disabled ]' /boot/user-data
+tail /boot/user-data
+
+# tell NetworkManager not to wait for the carrier on ethernet, which can delay boot
+# when the coprocessor isn't connected to the ethernet
+cat > /etc/NetworkManager/conf.d/50-ignore-carrier.conf <<EOF
+[main]
+ignore-carrier=*
```

Comment:
This might be dangerous on a robot network where ethernet cables can intermittently be unplugged.

Reply:
Can you share more details of the failure mode that you're concerned about? I'm testing different network setting to understand how they impact startup time and network behavior to try to find a configuration that provides the most stability and predictability, so understanding what could go wrong from your perspective would be a big help.